File: Rust/Medium/CompileError/course-schedule-iv_5.rs
impl Solution {
    pub fn check_if_prerequisite(num_courses: i32, prerequisites: Vec<Vec<i32>>, queries: Vec<Vec<i32>>) -> Vec<bool> {
        
        let num_courses = num_courses as usize;
        let mut graph = vec![vec![]; num_courses];
        for [a, b] in prerequisites {
            graph[a as usize].push(b as usize);
        }
        let mut reachable = vec![vec![false; num_courses]; num_courses];
        for node in 0..num_courses {
            let mut queue = vec![];
            queue.push(node);
            while let Some(front) = queue.pop() {
                for &neigh in &graph[front] {
                    if !reachable[node][neigh] {
                        reachable[node][neigh] = true;
                        queue.push(neigh);
                    }
                }
            }
        }
        queries.iter().map(|[a, b]| reachable[*a as usize][*b as usize]).collect()
    }
}

File: C++/Medium/Accepted/course-schedule-iv_5.cpp
class Solution {
public:
    vector<bool> checkIfPrerequisite(int numCourses, vector<vector<int>>& prerequisites, vector<vector<int>>& queries) {
        
        vector<vector<int>> g(numCourses, vector<int>(numCourses, 0));
        for (auto& p: prerequisites) {
            g[p[0]][p[1]] = 1;
        }
        for (int i = 0; i < numCourses; i++) {
            for (int j = 0; j < numCourses; j++) {
                for (int k = 0; k < numCourses; k++) {
                    if (g[j][i] && g[i][k]) {
                        g[j][k] = 1;
                    }
                }
            }
        }
        vector<bool> res;
        for (auto& q: queries) {
            res.push_back(g[q[0]][q[1]]);
        }
        return res;
    }
};